src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -1311,3 +1334,82 @@ double GetStreamCpuLoad( PaStream * s )
 
     return PaUtil_GetCpuLoad( &stream->cpuLoadMeasurer );
 }
+
+/** Extensions */
+static void RenameStreamCb(pa_stream *s, int success, void *userdata)
+{
+    /* Currently does nothing but signal the caller. */
+    PaPulseAudio_Stream *l_ptrStream = (PaPulseAudio_Stream *) userdata;
+    pa_threaded_mainloop_signal( l_ptrStream->mainloop,
+                                 0 );
+}
+
+PaError PaPulseAudio_RenameSource( PaStream *s, const char *streamName )
+{
+    PaPulseAudio_Stream *stream = (PaPulseAudio_Stream *) s;
+    PaError result = paNoError;
+    pa_operation *op = NULL;
+
+    /* Reallocate stream name in memory. */
+    char *newStreamName = (char*)PaUtil_AllocateMemory(strlen(streamName));
+    if ( !newStreamName )
+    {
+        return paInsufficientMemory;
+    }
+    strcpy(newStreamName, streamName);
+
+    PaUtil_FreeMemory( stream->sourceStreamName );
+    stream->sourceStreamName = newStreamName;
+
+    if ( stream->inStream == NULL )
+    {
+        return result;
+    }
+
+    pa_threaded_mainloop_lock( stream->mainloop );
+    op = pa_stream_set_name( stream->inStream, streamName, RenameStreamCb, stream );
+    pa_threaded_mainloop_unlock( stream->mainloop );
+
+    /* Wait for completion. */
+    while (pa_operation_get_state( op ) == PA_OPERATION_RUNNING)
+    {
+        pa_threaded_mainloop_wait( stream->mainloop );
+    }
+
+    return result;
+}
+
+PaError PaPulseAudio_RenameSink( PaStream *s, const char *streamName )
+{
+    PaPulseAudio_Stream *stream = (PaPulseAudio_Stream *) s;
+    PaError result = paNoError;
+    pa_operation *op = NULL;
+    
+    /* Reallocate stream name in memory. */
+    char *newStreamName = (char*)PaUtil_AllocateMemory(strlen(streamName));
```

Comment:
I think we should use strnlen like (paranoidd I know) with PAPULSEAUDIO_MAX_DEVICENAME just in case

Reply:
Updated to use strnlen/strncpy with PAPULSEAUDIO_MAX_DEVICENAME.